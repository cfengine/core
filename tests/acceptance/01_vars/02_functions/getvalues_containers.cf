body common control
{
      inputs => { "../../default.cf.sub" };
      bundlesequence  => { default("$(this.promise_filename)") };
      version => "1.0";
}

#######################################################

bundle agent init
{
  vars:
      "arr" data => '["a", [], { "x": 1 }, 2, 5.30, true]';
}

bundle agent test
{
  vars:
      "arr_v" slist => getvalues("init.arr");

reports:
  DEBUG::
    "arr_v $(arr_v)";
}

bundle agent check
{
  methods:
      "check"  usebundle => dcs_check_state(test,
                                           "$(this.promise_filename).expected.json",
                                           $(this.promise_filename));
}
