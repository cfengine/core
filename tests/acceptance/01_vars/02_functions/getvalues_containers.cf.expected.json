{
  "arr_v": [
    "a",
    "1",
    "2",
    "5.30",
    "true"
  ]
}
